### PR TITLE
CLOUDP-259871: Plugin discovery logic and plugin list command

### PR DESCRIPTION
**Pull request by stefan4h**
## Proposed changes

- Plugin discovery logic
- Add `plugin` sub command
- Add `plugin list` sub command

_Jira ticket:_ CLOUDP-259871

### CODE REVIEW

File: tools/cli-generator/generate.go [hunk 1]
```diff
@@ -237,7 +237,8 @@ func cleanupFile(filePath string) error {
 	if err != nil {
 		return err
 	}
-	return os.WriteFile(filePath, r, os.ModePerm)
+	const filePerm = 0600
```
Comment (jeroenvervaeke) [inline tools/cli-generator/generate.go:240]: Do we need this change as part of this PR?

**Review by jeroenvervaeke — CHANGES_REQUESTED** [review]

File: tools/cli-generator/generate.go (hunk 1 above)
Comment (jeroenvervaeke) [reply tools/cli-generator/generate.go:240]: I think you also changed this here: #3101

File: internal/plugin/plugin_manifest.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	}
+	if p.Description == "" {
+		errorsList = append(errorsList, fmt.Errorf(errorMessage, "description"))
+	}
+	if p.Binary == "" {
+		errorsList = append(errorsList, fmt.Errorf(errorMessage, "binary"))
+	}
+	if p.Version == "" {
+		errorsList = append(errorsList, fmt.Errorf(errorMessage, "version"))
+	} else if valid, _ := regexp.MatchString(`^\d+\.\d+\.\d+$`, p.Version); !valid {
```
[the remark this answers is not in the document]
Comment (stefan4h) [reply internal/plugin/plugin_manifest.go:55]: FYI we are using [this semver package](https://pkg.go.dev/github.com/Masterminds/semver/v3@v3.2.1) and not the one you posted so the validation looks slightly different:

```go
	} else if _, err := semver.NewVersion(p.Version); err != nil {
```

**Comment by gssbzn** [comment]
bad merge?

**Review by sarahsimpers — APPROVED** [review]
LGTM

File: tools/cli-generator/generate.go (hunk 1 above)
Comment (stefan4h) [reply tools/cli-generator/generate.go]: Yes I did, removed it from this PR now

**Review by jeroenvervaeke — APPROVED** [review]
LGTM

**Pull request merged**